plugin/pkg/scheduler/schedulercache/nodeinfocache/node_info_cache_test.go

```diff
@@ -0,0 +1,410 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeinfocache
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+)
+
+// TestAssumePod tests that after a pod is assumed, its information is aggregated
+// on node level.
+func TestAssumePod(t *testing.T) {
+	nodeName := "node"
+	tests := []struct {
+		pods []*api.Pod
+
+		wNodeInfo *schedulercache.NodeInfo
+	}{{
+		pods: []*api.Pod{makeBasePod(nodeName, "test", "100m", "500", []api.ContainerPort{{HostPort: 80}})},
+		wNodeInfo: &schedulercache.NodeInfo{
+			RequestedResource: &schedulercache.Resource{
+				MilliCPU: 100,
+				Memory:   500,
+			},
+			PodNum: 1,
+		},
+	}, {
+		pods: []*api.Pod{
+			makeBasePod(nodeName, "test-1", "100m", "500", []api.ContainerPort{{HostPort: 80}}),
+			makeBasePod(nodeName, "test-2", "200m", "1Ki", []api.ContainerPort{{HostPort: 8080}})},
+		wNodeInfo: &schedulercache.NodeInfo{
+			RequestedResource: &schedulercache.Resource{
+				MilliCPU: 300,
+				Memory:   1524,
+			},
+			PodNum: 2,
+		},
+	}}
+
+	for i, tt := range tests {
+		cache := newNodeInfoCache(time.Second, time.Second, nil)
+		for _, pod := range tt.pods {
+			err := cache.AssumePod(pod)
+			if err != nil {
+				t.Fatalf("AssumePod failed: %v", err)
+			}
+		}
+		n := cache.getNodeInfo(nodeName)
+		if !reflect.DeepEqual(n, tt.wNodeInfo) {
+			t.Errorf("#%d: node info get=%s, want=%s", i, n, tt.wNodeInfo)
+		}
+	}
+}
+
+// TestExpirePod tests that assumed pods will be removed if expired.
+// The removal will be reflected in node info.
+func TestExpirePod(t *testing.T) {
+	nodeName := "node"
+	now := time.Now()
+	ttl := 10 * time.Second
```

Comment:
IMHO The codebase has enough duplicated oddities that it's fair to say, "what is there got us working", but may not be what we want going fowards... So rationalizing the structure makes a lot of sense to me.  

+1 to decruft. 

/cc @davidopp